book/src/tutorial/sys_library.md

```diff
@@ -1,81 +1,45 @@
-cargo new sourceview --lib
-```
-
-Then let's create a folder inside the newly created `sourceview` folder for the `sys` part:
+We have installed [gir], set up our repo and have all the `.gir` files we need. Now let's work on the unsafe bindings of the -sys crate. Let's change into the directory of the sys crate.
 
 ```sh
-cd sourceview
-cargo new sourceview-sys --lib
+cd pango/pango-sys
 ```
 
-To indicate to [gir] what to generate, we'll need a `Gir.toml` file (inside the `sourceview-sys` folder) containing:
+## The Gir.toml file
+The first step is to let [gir] know what to generate. This is what the `Gir.toml` file that we created inside the `pango-sys` folder is for. This file will not be replaced when we run gir again. The file is currently empty so let's add the following to it:
 
 ```toml
 [options]
-library = "GtkSource"
-version = "3.0"
+library = "Pango"
+version = "1.0"
+min_cfg_version = "1.0"
 target_path = "."
-min_cfg_version = "3.0"
+girs_directories = ["../../gir-files/"]
+work_mode = "sys"
+single_version_file = true
 ```
 
-* `library` stands for the library we want to generate.
-* `version` stands for the version of the library to be used.
-* `target_path` stands for the location where the files will be generated.
+* `library` stands for the name of the library we want to generate.
+* `version` stands for the version of the library to be used. 
 * `min_cfg_version` will be the minimum version supported by the generated bindings.
+* `target_path` stands for the location where the files will be generated.
+* `girs_directories` stands for the location of the `.gir` files.
+* `work_mode` stands for the mode gir is using. The options here are `sys` and `normal`.
+* `single_version_file` is a very useful option when you have a lot of generated files (like we'll have). Instead of generating the gir hash commit used for the generation in the header of all generated files, it'll just write it inside one file, removing `git diff` noise **a lot**.
 
-You should now have a folder looking like this:
-
-```text
-sourceview/
-  |
-  |---- Cargo.toml
-  |---- sourceview-sys/
-  |       |
-  |       |---- Cargo.toml
-  |       |---- Gir.toml
-  |       |---- src/
-  |               |
-  |               |---- lib.rs
-  |---- src/
-          |
-          |---- lib.rs
-```
+You can find out the values for `library` and `version` by looking at the name of the .gir file of your library. In our case it is called Pango-1.0.gir. This tells us that the `library` is Pango and the `version` is 1.0. If you don't know what value to use for `min_cfg_version`, use the same as you use for `version`. If not all `.gir` files that are needed reside in `../../gir-files/`, you can add the path to the other files by changing `girs_directories`. If for example you also have `.gir` files in the root of your project folder, change it to `girs_directories = ["../../gir-files/", "../.."]`. Because we are generating the unsafe bindings, we use the `sys` work mode.
```

Comment:
```suggestion
You can find out the values for `library` and `version` by looking at the name of the .gir file of your library. In our case it is called Pango-1.0.gir. This tells us that the `library` is Pango and the `version` is 1.0. If you don't know what value to use for `min_cfg_version`, use the same as you use for `version`. If not all needed `.gir` files reside in `../../gir-files/`, you can add the path to the other files by changing `girs_directories`. If for example you also have `.gir` files in the root of your project folder, change it to `girs_directories = ["../../gir-files/", "../.."]`. Because we are generating the unsafe bindings, we use the `sys` work mode.
```